### PR TITLE
Add an example for FromIterator::from_iter

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by Manishearth** [comment]
This looks good to me. r=me with or without the following change:

While we're at it, could we have an example that uses `collect()`? That is the more common use for `FromIterator`, eg `(vec![1,2,3,4]).map(|x| x*x).collect::<Vec<_>>()`

**Comment by frewsxcv** [comment]
@Manishearth How's that?

File: src/libcore/iter.rs [hunk 1]
```diff
@@ -1017,6 +1017,28 @@ impl<'a, I: Iterator + ?Sized> Iterator for &'a mut I {
...
+    /// `FromIterator` is more commonly used implicitly via the `Iterator::collect` method:
+    ///
+    /// ```
+    /// use std::collections::HashSet;
+    ///
+    /// let colors_vec = vec!["red", "red", "yellow", "blue"];
+    /// let colors_set = colors_vec.into_iter().collect::<HashSet<&str>>();
+    /// assert_eq!(colors_set.len(), 3);
+    /// ```
+    assert_eq!(colors_set.len(), 3);
```
Comment (Manishearth) [inline src/libcore/iter.rs]: this line seems to be a mistake?

Comment (frewsxcv) [inline src/libcore/iter.rs]: Yep. Fixed in latest push

**Pull request merged**